### ISSUE TITLE
mock-fs does not work with babel

### ISSUE THREAD

**Issue by city41** [tschaub/mock-fs#49]
When using mock-fs 3.0.0 with babel, you get this error:

```
Message:
    /Users/matt/dev/get-nrdp/node_modules/mock-fs/node/fs-0.12.0.js:1
te = global.clearImmediate; var console = global.console; var __core-js_shared
                                                                    ^
Unexpected token -
Stack:
/Users/matt/dev/get-nrdp/node_modules/mock-fs/node/fs-0.12.0.js:1
te = global.clearImmediate; var console = global.console; var __core-js_shared
                                                                    ^
SyntaxError: Unexpected token -
    at exports.runInThisContext (vm.js:73:16)
    at Module._compile (module.js:443:25)
    at Module._extensions..js (module.js:478:10)
    at Object.require.extensions.(anonymous function) [as .js] (/Users/matt/dev/get-nrdp/node_modules/babel/node_modules/babel-core/lib/api/register/node.js:214:7)
    at Module.load (module.js:355:32)
    at Object.load (/Users/matt/dev/get-nrdp/node_modules/mock-fs/node_modules/rewire/lib/moduleEnv.js:19:18)
    at internalRewire (/Users/matt/dev/get-nrdp/node_modules/mock-fs/node_modules/rewire/lib/rewire.js:56:15)
    at rewire (/Users/matt/dev/get-nrdp/node_modules/mock-fs/node_modules/rewire/lib/index.js:11:12)
    at Object.<anonymous> (/Users/matt/dev/get-nrdp/node_modules/mock-fs/lib/index.js:42:14)
    at Module._compile (module.js:460:26)
```

This is because mock-fs is using rewire 2.0.0. rewire 2.0.0 did not correctly handle the `global.__core-js_shared` property, it tries to do `var __core-js-shared = ...`

rewire 2.3.0 addresses this issue. I forked mock-fs, upgraded to rewire 2.3.0 and now everything works great.

I'd send you a pull request, but I'm unsure if rewire 2.3.0 causes other issues.

**Comment by dave-irvine** [comment]
+1 just run into this as well :(

**Comment by city41** [comment]
@dave-irvine feel free to use my fork: https://github.com/city41/mock-fs.git

The only difference is the rewire upgrade. Have been using it with babel with no issues.

**Comment by dave-irvine** [comment]
@city41 Yeah this is basically what I did. I haven't had any issues either, so is it worth you doing the pull request you mentioned?

**Comment by tschaub** [comment]
Looks like this should have been fixed by #60.  Please reopen if not.
